File: src-tauri/src/models/comment.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct NodeComment {
    pub comment_id: String,
    pub project_id: String,
    pub node_id: String,
    pub iteration_id: String,
    pub json_path: String,
    pub comment_text: String,
    pub author: String,
    pub is_resolved: bool,
    pub created_at: String,
    pub updated_at: String,
}

File: src/types/comment.ts
export interface NodeComment {
  comment_id: string;
  project_id: string;
  node_id: string;
  iteration_id: string;
  json_path: string;
  comment_text: string;
  author: string;
  is_resolved: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateCommentParams {
  projectId: string;
  nodeId: string;
  iterationId: string;
  jsonPath: string;
  commentText: string;
}
